watcher.go

```diff
@@ -32,23 +35,52 @@ const (
 
 type Watcher interface {
 	GetQueries() []CarQuery
+	GetFlexCarQuery() CarQuery
 	OnCarAvailable(query CarQuery, cars []Car)
+	OnFlexCarAvailable(cars []FlexCarAvailabilityResp)
 }
 
 type WatcherOptions struct {
-	Interval time.Duration
-	Watcher  Watcher
+	Interval        time.Duration
+	Watcher         Watcher
+	IsFetchStations bool
+	IsFetchFlexCars bool
```

Comment:
The `CarQuery` could specify if it's a flex or a station query. We wouldn't need both `GetQueries` + `GetFlexCarQuery` to be defined on the Watcher struct, as well as those IsFetchStations + IsFetchFlexCars fields